std/file.d

```diff
@@ -4308,7 +4307,24 @@ string tempDir() @trusted
         }
         else static assert(false, "Unsupported platform");
 
-        if (cache is null) cache = getcwd();
+        if (cache is null)
+        {
+            import std.path : dirSeparator;
+            cache = getcwd() ~ dirSeparator;
+        }
     }
     return cache;
 }
+
+@trusted unittest
+{
+    version(Windows)
+    {
+        enum pathSeparator = '\\';
+    }
+    else
+    {
+        enum pathSeparator = '/';
```

Comment:
Why not use `std.path.dirSeparator`, as in the code you added above? That would allow redeclaring the constant, and the `version` blocks.